### PR TITLE
fix(vscode): reopen files that were open before diff view

### PR DESCRIPTION
**Pull request by zhanba**
## Summary
- Fixes issue where files open in VSCode before creating a diff view would remain closed after accepting or rejecting changes
- Tracks whether a file was open before the diff view via `wasFileOpenBeforeDiff` property
- Automatically reopens the file after applying changes if it was previously open
- Adds `isFileCurrentlyOpen` utility to detect open files (excluding diff views)

This improves the user experience by preserving their workspace state when working with diff views.

## Changes
- Added `wasFileOpenBeforeDiff` property to DiffView class to track file state
- Modified DiffView constructor to accept and store the open file state
- Enhanced `acceptChanges` method to conditionally reopen files
- Added `isFileCurrentlyOpen` utility function to check if file is open in tabs
- Updated `DiffView.create` to detect and pass the current file state

## Test plan
- [x] Verify that files open before diff view are reopened after accepting changes
- [x] Verify that files not previously open remain closed after diff operations  
- [x] Test with various file types and edge cases
- [x] Ensure no regression in existing diff view functionality

🤖 Generated with [Pochi](https://getpochi.com)

### CODE REVIEW

File: packages/vscode/src/integrations/editor/diff-view.ts [hunk 1]
```diff
@@ -39,6 +39,7 @@ export class DiffView implements vscode.Disposable {
     private readonly fileExists: boolean,
     private readonly originalContent: string,
     private readonly activeDiffEditor: vscode.TextEditor,
+    private readonly wasFileOpenBeforeDiff = false,
```
Comment (wsxiaoys) [inline packages/vscode/src/integrations/editor/diff-view.ts]:
```suggestion
    private readonly isFileOpenBeforeDiffPreview = false,
```

File: packages/vscode/src/integrations/editor/diff-view.ts [hunk 2]
```diff
@@ -400,6 +422,22 @@ export class DiffView implements vscode.Disposable {
   };
 }
 
+// Check if a file is currently open in any tab (excluding diff views)
+async function isFileCurrentlyOpen(fileUri: vscode.Uri): Promise<boolean> {
```
Comment (wsxiaoys) [inline packages/vscode/src/integrations/editor/diff-view.ts]:
```suggestion
async function isFileOpen(fileUri: vscode.Uri): Promise<boolean> {
```

**Review by wsxiaoys — APPROVED** [review]
a few nits, otherwise LGTM
